File: leetcode/go/3 Longest Substring Without Repeating Characters.go
func max(a int, b int) int {
    if(a > b){
        return a
    }else{
        return b
    }
}

func lengthOfLongestSubstring(s string) int {
    mp := make(map[byte]byte)
    i := 0
    res := 0
    left := 0
    
    for; i < len(s);{
        _, ok := mp[s[i]]
        
        if(ok){
            delete(mp, s[left])
            left++
        }else{
            mp[s[i]] = s[i]
            i++
            res = max(res, len(mp))
        }
    }
    
    return res
}

File: leetcode/ts/3 Longest Substring Without Repeating Characters.ts
function lengthOfLongestSubstring(s: string): number {
    let res : number = 0
    let right : number = 0
    let left : number = 0
    let map : Map<any, boolean> = new Map()
    
    while(right < s.length){
        if(!map.has(s[right])){
            map.set(s[right], true)
            res = Math.max(res,  right - left + 1 )
            ++right
        }else{
            map.delete(s[left++])                         
        }
    }
    return res
};